sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -98,6 +111,18 @@ func Main(ctx context.Context, loggingEndpoint, controlEndpoint string) error {
 		log.Debugf(ctx, "control response channel closed")
 	}()
 
+	// if the runner supports worker status api then expose SDK harness status
+	if statusEndpoint != "" {
+		statusHandler, err := newWorkerStatusHandler(ctx, statusEndpoint)
+		if err != nil {
+			log.Errorf(ctx, "error establishing connection to worker status API: %v", err)
+		} else {
+			statusHandler.wg.Add(1)
```

Comment:
Consider calling this in the `start` method instead. It should go right before the `go w.reader(...)` call. It can't be called in the `reader` method because it's being run on a different goroutine, and there's no guarantee that it will run ASAP (though that is true in practice). Calling it before, ensures that there shouldn't be a race condition on something to wait on, before it's time to stop.

(Unless I'm missing something of course, at which point we should have a comment here explaining why we need to do the wg.Add here)

Reply:
Makes sense